design/protected-apps-view.md

```diff
@@ -0,0 +1,917 @@
+Stale Data: Browser-side correlation creates a window where UI shows inconsistent state (e.g., ApplicationSet visible but no DR info shown), causing confusion about protection status. Users frequently refresh to resolve inconsistencies.
+Limited Filtering: Client-side filtering requires fetching and correlating all applications first. No efficient queries like "all ApplicationSets protected by metro-dr-policy."
+
+---
+
+## 4. Proposed Solution: Controller-Based Aggregation
+
+### 4.1 Design Philosophy
+
+The proposed solution moves the correlation logic from the frontend to the backend, implementing it as Kubernetes controllers that maintain an aggregated view resource. This architectural shift is based on several key principles that address the limitations we've identified.
+
+Rather than having correlation logic scattered across multiple frontend hooks, we create a single authoritative source of truth: a new Custom Resource Definition called ProtectedApplicationView. This CRD represents the aggregated information about a protected application, regardless of its type. Every protected application gets one corresponding ProtectedApplicationView resource, and the frontend simply watches and displays these views.
+
+This approach provides strong consistency guarantees. When a user sees a ProtectedApplicationView resource, they know it reflects the current state of the underlying application and its DR configuration. There are no race conditions between multiple hooks trying to correlate the same data independently.
+
+
+### 4.2 Architecture Overview
+
+The proposed architecture introduces three new controllers and one new CRD. Let's examine each component and how they work together.
+
+#### 4.2.1 The ProtectedApplicationView CRD
+
+This new custom resource represents the aggregated view of a protected application. Its schema is designed to accommodate all three application types while making the common patterns easy to work with:
+
+```yaml
+apiVersion: ramendr.openshift.io/v1alpha1
+kind: ProtectedApplicationView
+metadata:
+  name: appset-payment-service  # Prefixed with type for uniqueness
+  namespace: openshift-gitops    # Same namespace as the source application
+  
+  # Owner reference ensures automatic cleanup
+  ownerReferences:
+    - apiVersion: argoproj.io/v1alpha1
+      kind: ApplicationSet
+      name: payment-service
+      uid: <applicationset-uid>
+      controller: true
+      blockOwnerDeletion: true
+  
+  # Labels enable efficient querying
+  labels:
+    app.kubernetes.io/type: ApplicationSet
+    ramendr.openshift.io/dr-policy: metro-dr-policy
+    ramendr.openshift.io/protected: "true"
+
+spec:
+  # The type determines which fields are populated
+  type: ApplicationSet  # or "Subscription" or "Discovered"
+  
+  # Reference back to the actual application resource
+  applicationRef:
```

Comment:
we also have a spec `drpcRef` in this CR, for discovered apps isn't `applicationRef` == `drpcRef` and kind of redundant to have same info twice ??